---
title: "DESEQ2_DATASET1"
author: "Ryan Brownlee"
date: '2022-06-30'
output: html_document
---
### Purpose  
This markdown uses nf-core for RNAseq alignment and DESeq2 DDS output  
Dataset: [GSE149739](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149739)  
Processed data filepath: /data/project/lasseigne_lab/Lizzy/PKD_drugrepurposing_220324/  

#### Load in libraries
```{r message=FALSE, warning = FALSE}
library(DESeq2)
library(stringr)
library(apeglm)
library(pheatmap)
library(dplyr)
library(ggrepel)
library(signatureSearch)
library(ExperimentHub); library(rhdf5)
library(biomaRt)
library(viridis)
```

#### Custom functions  

__salmon2dds__: Salmon merged gene counts from nf-core RNAseq pipeline (results/star_salmon/salmon.merged.gene_counts.rds) and SRA metadata as inputs, function data wrangles and runs DESeq2, resulting in a dds object with design by genotype    

```{r}
#Salmon merged gene counts to dds results
salmon2dds <- function(counts, metadata) {
  #count matrix in the sumarized experiment object
  cts <- counts@assays@data@listData[["counts"]]
  #round, salmon outputs have decimals
  cts <- round(cts)
  #colnames(cts) <- str_sub(colnames(cts), 1, -2) ##ALTERATION just for GSE69556
  #remove version numbers from transcript id's
  rownames(cts) <- gsub("\\..*", "", rownames(cts))
  
  #remove all non-alphanumeric characters from Genotype
  metadata$Genotype <- str_replace_all(metadata$Genotype, "[^[:alnum:]]", "")
  #contrasts need to be factor levels
  metadata$Genotype <- as.factor(metadata$Genotype)
#print(rownames(metadata))
#print(colnames(cts))
  dds <- DESeqDataSetFromMatrix(cts, metadata, design = ~ Genotype)
return(dds)
}
```

__uplfc__ & __downlfc__: Formats deseq2 res() output to select for upregulated and downregulated genes by log2 fold change threshold > 2 and < -2.  

```{r}
uplfc <- function(res){
  #NAs removed first
  res <- res[!is.na(res$log2FoldChange),]
  #p-adjusted value cutoff 0.05 and log2fc greater than 2.0 and less than -2.0
  upfc <- res[res$log2FoldChange > 2.0,]
  #pull just logfold change values from DESeqResults object and name the values by associated gene
  up <- upfc$log2FoldChange
  #names(up) <- rownames(upfc)
  up <- cbind(Mouse.gene.stable.ID = rownames(upfc), UpLFC = up)
  
return(as.data.frame(up))
}
downlfc <- function(res){
  #NAs removed first
  res <- res[!is.na(res$log2FoldChange),]
  #p-adjusted value cutoff 0.05 and log2fc greater than 2.0 and less than -2.0
  downfc <- res[res$log2FoldChange < -2.0,]
  #pull just logfold change values from DESeqResults object and name the values by associated gene
  down <- downfc$log2FoldChange
  #names(down) <- rownames(downfc)
  down <- cbind(Mouse.gene.stable.ID = rownames(downfc), DownLFC = down)
  
return(as.data.frame(down))
}
```

__convertMouseGeneList__: Uses Ensembl human GRCh38.p13 and Ensembl mouse C57BL_6NJ_v1 genes and to find human orthologous genes (more details [here](https://docs.google.com/document/d/1jU3EOVaZXJMzxwH9SsssHoArz8pMdcfW8MSr8hbeE4Q/edit) ), and biomart to convert to human entrez  

```{r}
convertMouseGeneList <- function(lfc){
  #read in mose to human ens annotation
  ens_annot <- read.csv("/Users/eramsey/Downloads/PKD/kidney_drugrepurposing/data/annot_ens_humanmouse.csv")
  #conversion table
  ens_mouse2human <- merge(lfc, ens_annot, by = "Mouse.gene.stable.ID")
  #biomart to conver human ens to human entrez
  human <- useEnsembl("ensembl", dataset = "hsapiens_gene_ensembl", mirror = "useast")
  genes <- getBM(filters = "ensembl_gene_id", attributes = c("ensembl_gene_id","entrezgene_id"), values = ens_mouse2human$Gene.stable.ID, mart = human)
  
  names(ens_mouse2human) <- c("mouse_ensembl_gene_id", "LFC", "ensembl_gene_id")
  conv_tbl <- merge(genes, ens_mouse2human, by = "ensembl_gene_id")
return(conv_tbl)
}
```

#### Read in data  
Filepath to counts data: /data/project/lasseigne_lab/Lizzy/PKD_drugrepurposing_220324/  
Metadata: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE149739 

```{r}
cts_39 <- readRDS("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\Gene_counts_Database1.rds")
meta_39 <- read.csv("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\MetaData_Database1.txt", row.names = 1)
```

### DDS
```{r}
dds_39 <- salmon2dds(cts_39, meta_39)
```


#### Run DESeq2
```{r}
#relevel to make wildtype the reference instead of first alphabetical factor
dds_39$Genotype <- relevel(dds_39$Genotype, ref = "Wildtype")
#run DESeq: estimates size factors, estimates dispersion, fits negative binomial GLM's and performs Wald stats
dds_39 <- DESeq(dds_39) #can also look at setting replaceOutliers lower for dealing with outliers flagged by extreme Cook's distances
#save(ds, file = "./res/deseq2_outputs/GSE149739_dds_kasi2021alignment.rds")
```

#### Results  
Note: for LFC, if using apeglm for shrinkage estimate prior: If used in published research, please cite:  
Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for sequence count data: removing the noise and preserving large differences. Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
```{r}
#extract results table with log2 fold changes, p values and adjusted p values for desired coefficients
#alpha 0.05 cutoff
res_nf <- results(dds_39, alpha = 0.05, contrast = c("Genotype", "Pkd2flflPax8rtTATetOcre", "Wildtype"))
#Shrinkage of effect size (LFC estimates) is useful for visualization and ranking of genes. To shrink the LFC, we pass the ddds_39_nf object to the function lfcShrink
resultsNames(dds_39) #prints coefficients
# "type" refers to different types of priors for shrinkage estimator that can be used: apeglm, ashr, or normal
## apeglm and ashr have been shown to have less bias than normal
resLFC_nf <- lfcShrink(dds_39, coef = c("Genotype_Pkd2flflPax8rtTATetOcre_vs_Wildtype"), type="apeglm")
resLFC_nf
```

DESeq2 results
```{r}
dsq2_39 <- as.data.frame(res_nf)
dsq2_39_up <- dplyr::filter(dsq2_39, log2FoldChange > 2.0)
dsq2_39_down <- dplyr::filter(dsq2_39, log2FoldChange < -2.0)
```

Save full deseq results
```{r}
write.csv(dsq2_39, file = "./res/deseq2_outputs/deseq2_fullres_39.csv")
write.csv(dsq2_39_up, file = "./res/deseq2_outputs/deseq2_upgenes_res_39.csv")
write.csv(dsq2_39_down, file = "./res/deseq2_outputs/deseq2_downgenes_res_39.csv")
```

#### LFC cutoffs
Using < -2 and > 2 LFC threshold for up and downregulated genes 
```{r}
musdegs_39_UP <- uplfc(res_nf)
head(musdegs_39_UP)
musdegs_39_DOWN <- downlfc(res_nf)
head(musdegs_39_DOWN)
```


Variance stabilized transformation  
vst() uses a parametric fit for the dispersion. In this case, the closed-form expression for the variance stabilizing transformation is used by the vst function. If a local fit is used (option fitType="locfit" to estimateDispersions) a numerical integration is used instead. The transformed data should be approximated variance stabilized and also includes correction for size factors or normalization factors. The transformed data is on the log2 scale for large counts.  
  
rlog() is better to use for this dataset -- better for high count variance by sample (in this case, one sample has over 2x the reads compared to the others) and for smaller datasets "if you have many samples (e.g. 100s), the rlog function might take too long, and so the vst function will be a faster choice. The rlog and VST have similar properties, but the rlog requires fitting a shrinkage term for each sample and each gene which takes time. See the DESeq2 paper for more discussion on the differences (Love, Huber, and Anders 2014)."  
```{r}
#vsd_nf <- vst(dds_39, blind = FALSE) #if blind, could minimize large count differences due to anticipated experimental variation
rld_nf <- rlog(dds_39, blind = FALSE) 
```


#### Visualizations  
MA plot:  
In DESeq2, the function plotMA shows the log2 fold changes attributable to a given variable over the mean of normalized counts for all the samples in the DESeqDataSet. Points will be colored red if the adjusted p value is less than 0.1. Points which fall out of the window are plotted as open triangles pointing either up or down.  
It is more useful visualize the MA-plot for the shrunken log2 fold changes, which remove the noise associated with log2 fold changes from low count genes without requiring arbitrary filtering thresholds.  
```{r}
plotMA(resLFC_nf, ylim=c(-2,2))
```

PCA
```{r}
#Plot PCA
plotPCA(rld_nf, intgroup=c("Genotype")) + geom_label_repel(aes(label = rld_nf@colData@rownames), box.padding   = 0.15, label.padding = 0.15, point.padding = 0.3, segment.color = 'grey50') + scale_colour_viridis_d() + theme(axis.title.x = element_text(size = 20)) + theme_minimal()

#PCATOOLS ATTEMPT
newdf <- rownames(cts_39)
newdf <- mapIds(org.Mm.eg.db, cts_39@NAMES, keytype = "ENSEMBL", column = "SYMBOL", multiVals = "first")
newdf <- newdf[!is.na(newdf)]
newdf <- newdf[match(rownames(cts_39), names(newdf))]
rownames(cts_39) <- newdf
keep <- !is.na(rownames(cts_39))
cts_39 <- cts_39[keep,]
dds <- DESeqDataSet(cts_39, design = ~ cell + dex)

```

Modify DESeq2::plotPCA
```{r}
plotPCA2 <- function (object, intgroup = "condition", ntop = 1000, returnData = FALSE, PC1, PC2) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = " : "))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC1 = pca$x[, 1], PC2 = pca$x[, 2], PC3 = pca$x[, 3], PC4 = pca$x[, 4], PC5 = pca$x[,5], PC6 = pca$x[,6], PC7 = pca$x[,7], PC8 = pca$x[,8], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[1:2]
        return(d)
    }
    ggplot(data = d, aes_string(x = PC1, y = PC2, color = "group")) + 
        geom_point(size = 3) + xlab(paste0(as.character(PC1), ": ", round(percentVar[1] * 
        100), "% variance")) + ylab(paste0(as.character(PC2), ": ", round(percentVar[3] * 
        100), "% variance")) + coord_fixed()
}
```
### Mouse to Human Ortholog Mapping
Map to human entrez and filter by LINCS genes
```{r}
hom_degs_UP <- lincsGenes(musdegs_39_UP, "up")
hom_degs_DOWN <-  lincsGenes(musdegs_39_DOWN, "down")
```


Orthologous mapping highest weighted genes in PC1
```{r eval=FALSE, include=FALSE}
#orthologous mapping
#hom_pc1UP <- convertMouseGeneList(rownames(genesPC1_UP))
#hom_pc1DOWN <- convertMouseGeneList(rownames(genesPC1_DOWN))
```

Heatmap plotting:
#Top 20 LFC
```{r}
library(pheatmap)
#formatting for pheatmap
conversion <- as.data.frame(cts_39@elementMetadata)
conversion$gene_id <- gsub("\\..*", "", conversion$gene_id)
cts_df <- as.data.frame(assays((rld_nf)))
cts_df <- tibble:::rownames_to_column(cts_df, var = "gene_id")

conv_cts <- merge(cts_df, conversion, by = "gene_id")
conv_cts <- conv_cts[,!(colnames(conv_cts) %in% c("gene_id", "group", "group_name"))]
conv_cts <- dplyr::distinct(conv_cts, gene_name, .keep_all = TRUE)
conv_cts <- tibble:::column_to_rownames(conv_cts, "gene_name")
colnames(conv_cts) <- c("WT", "WT", "WT", "PKD2", "PKD2", "PKD2", "Ift88", "Ift88", "Ift88")
conv_cts[conv_cts==0] <- NA
conv_cts_fixed <- conv_cts[complete.cases(conv_cts),]
#conv_cts <- round(conv_cts)

#need to convert the gene IDs here also
#genes <- order(resLFC_nf$log2FoldChange, decreasing=TRUE)[1:20]

#top20 genes w/ conv_cts
fixedtop20 <- order(conv_cts_fixed$PKD2, decreasing = TRUE)[1:20]
#fixedtop20test <- order(conv_cts_fixed, decreasing= TRUE)[1:20]
pheatmap(conv_cts_fixed[fixedtop20, ], cluster_rows = TRUE, cluster_cols = TRUE, show_rownames = TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "Top 20 Positive Genes '39", angle_col = 45)

#bottom 20 genes w/ conv_cts
fixedbottom20 <- order(conv_cts_fixed$PKD2, decreasing = FALSE)[1:20]
pheatmap(conv_cts_fixed[fixedbottom20, ], cluster_rows = TRUE, cluster_cols = TRUE, show_rownames = TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "Bottom 20 Positive LFC", angle_col = 45)

#sort by variance
test <- apply(conv_cts, 1, var)
#top 20 most variable genes
top <- names(sort(test, decreasing = TRUE))[1:20]
#create heatmap
pheatmap(conv_cts[top,], cluster_rows=TRUE, cluster_cols=TRUE, show_rownames=TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "20 Most Varied Genes '39", angle_col = 45) #, annotation_col=meta_39$Genotype)
           
   #heatmap w/o gene symbols 20 greatest genes
genestop20 <- order(resLFC_nf$log2FoldChange, decreasing = TRUE)[1:20]
pheatmap(assay(cts_39)[genestop20, ], cluster_rows = TRUE, show_rownames = TRUE, cluster_cols = TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "20 Greatest LFC Genes '39", angle_col = 45)

#Heatmap w/o gene symbols 20 lowest genes
genesbottom20 <- order((resLFC_nf$log2FoldChange), decreasing = FALSE)[1:20]
pheatmap(assay(cts_39)[genesbottom20, ], cluster_rows = TRUE, show_rownames = TRUE, cluster_cols = TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "20 Lowest LFC Genes", angle_col = 45)
```
#heatmaps w/o IFT88
```{r}
conv_cts_fixed_newheat <- conv_cts_fixed[,!(colnames(conv_cts_fixed) %in% c("Ift88"))]
colnames(conv_cts_fixed_newheat) <- c("WT", "WT", "WT", "PKD2", "PKD2", "PKD2")
#top20 genes w/ conv_cts
fixedtop20 <- order(conv_cts_fixed_newheat$PKD2, decreasing = TRUE)[1:20]
#fixedtop20test <- order(conv_cts_fixed, decreasing= TRUE)[1:20]
pheatmap(conv_cts_fixed_newheat[fixedtop20, ], cluster_rows = TRUE, cluster_cols = TRUE, show_rownames = TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "Top 20 Positive Genes '39", angle_col = 45)

#bottom 20 genes w/ conv_cts
fixedbottom20 <- order(conv_cts_fixed_newheat$PKD2, decreasing = FALSE)[1:20]
pheatmap(conv_cts_fixed_newheat[fixedbottom20, ], cluster_rows = TRUE, cluster_cols = TRUE, show_rownames = TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "Bottom 20 Positive Genes '39", angle_col = 45)

#sort by variance
test <- apply(conv_cts, 1, var)
#top 20 most variable genes
top <- names(sort(test, decreasing = TRUE))[1:20]
#create heatmap
pheatmap(conv_cts_fixed_newheat[top,], cluster_rows=TRUE, cluster_cols=TRUE, show_rownames=TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "20 Most Varied Genes Pre-Cyst", angle_col = 45) #, annotation_col=meta_39$Genotype)
```


Plotting alluvial plot
```{r}
gene_targets <- read.csv("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\gene_drug_targets_39.csv")
gene_target_alluvial <- gene_targets[,c(2,19)]
gene_target_alluvial <- gene_target_alluvial[c(1,5,15,16,25,49,51,59,62,63,71,76,84,90,93,99,114,116,123,132,144,151,157),]
names(gene_target_alluvial)[1] <- "Drug"
library(ggplot2)
library(ggalluvial)
target_alluvial_plot <- ggplot(data = gene_target_alluvial, aes(axis1 = Drug, axis2 = Target)) +
  geom_alluvium(color = "black", aes(fill = Target )) + # , colour = trend)) + 
  geom_stratum(width = 3/12, color = "black") + #the border of the stratum
  scale_x_discrete(expand = c(.07, .07)) + 
  scale_y_discrete(expand = c(.007, .007)) + 
  scale_fill_viridis_d(aesthetics = c("colour", "fill")) +
  geom_text(stat = "stratum", size = 3.5, fontface = "bold", aes(label = after_stat(stratum))) +
  labs(title = "Drug Target Matches") +
  theme_void() 
target_alluvial_plot

```
#Individual Gene Targets
```{r}
individual_test <- plotCounts(dds_39, gene = "Ckb", intgroup = "Genotype", returnData = TRUE) 
```

```{r}
#volcano plotting
 if (!requireNamespace('BiocManager', quietly = TRUE))
    install.packages('BiocManager')

  BiocManager::install('EnhancedVolcano')
  library(EnhancedVolcano)
  EnhancedVolcano(res_nf, lab = rownames(res_nf), x = 'log2FoldChange', y = 'pvalue', pCutoff = 10e-32, FCcutoff = 0.5, pointSize = 3.0, labSize = 6.0)
  
TopT <- as.data.frame(res_nf)
TopT_test <- tibble:::rownames_to_column(TopT)
names(TopT_test)[1] <- "gene_id"
TopT_finish <- left_join(TopT_test, conversion, by = "gene_id")
TopT_finish <- na.omit(TopT_finish)
TopT_finish <- TopT_finish[,!(colnames(TopT_finish) %in% c("gene_id", "baseMean", "lfcSE", "stat"))]
TopT_finish <- dplyr::distinct(TopT_finish, gene_name, .keep_all = TRUE)
TopT_finish <- tibble:::column_to_rownames(TopT_finish, "gene_name")

View(TopT_finish)
#volcano plot w/ strict pvalue cutoff
EnhancedVolcano(TopT_finish, lab = rownames(TopT_finish), x = 'log2FoldChange', y = 'pvalue', pCutoff = 10e-32, FCcutoff = 0.5, pointSize = 3.0, labSize = 6.0, col = c("#440154", "#21918c", "#5ec962", "#fde725"))
#basic volcano plot
EnhancedVolcano(TopT_finish, lab = rownames(TopT_finish), x = 'log2FoldChange', y = 'pvalue', pointSize = 3.0, labSize = 4.0, col = c("#440154", "#21918c", "#5ec962", "#fde725"))
```

```{r}
# Read in Drug Targets
gene_targets <- read.csv("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\gene_drug_targets_39.csv")
gene_targets_unique <- data.frame(gene_targets$pert, gene_targets$Target)
gene_targets_unique <- setNames(gene_targets_unique, c("Drug", "gene_name"))
#Check for DEG drug targets
conversion <- as.data.frame(cts_39@elementMetadata)
conversion$gene_id <- gsub("\\..*", "", conversion$gene_id)
deg_39_up <- dsq2_39_up
deg_39_up <- tibble:::rownames_to_column(deg_39_up)
names(deg_39_up)[1] <- "gene_id"
deg_39_up <- left_join(deg_39_up, conversion, by = "gene_id")
deg_39_up <- deg_39_up[,!(colnames(deg_39_up) %in% c("gene_id", "baseMean", "lfcSE", "stat"))]
deg_39_up <- dplyr::distinct(deg_39_up, gene_name, .keep_all = TRUE)
#deg_39_up <- tibble:::column_to_rownames(deg_39_up, "gene_name")

# deg_39_up is in lowercase and gene_targets_unique_df is uppercase. must fix!
deg_39_up$gene_name <- toupper(deg_39_up$gene_name)
#now fixed, merge for similarities in targets
target_gene_overlap_DESEQ2_39 <- merge(deg_39_up, gene_targets_unique, by = "gene_name")
View(target_gene_overlap_DESEQ2_39)
write.csv(target_gene_overlap_DESEQ2_39, file = "C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\target_gene_overlap_DESEQ2_39.csv")
```


```{r}
Limma_Drug_Targets <- read.csv("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\rhub_limma_launcheddrugs_39.csv")
#separate gene targets into individual rows
Limma_Drug_Targets <- tidyr::separate_rows(Limma_Drug_Targets, Target, sep = ", ")
#remove drugs with empty 'target' field
Limma_Drug_Targets <- dplyr::filter(Limma_Drug_Targets, Target != "")
#Make new dataframe of just drug and gene name
Limma_Drug_Targets_unique <- data.frame(Limma_Drug_Targets$pert, Limma_Drug_Targets$Target)
#rename for merging function
Limma_Drug_Targets_unique <- setNames(Limma_Drug_Targets_unique, c("Drug", "gene_name"))
conversion <- as.data.frame(cts_39@elementMetadata)
conversion$gene_id <- gsub("\\..*", "", conversion$gene_id)
deg_39_up <- dsq2_39_up
deg_39_up <- tibble:::rownames_to_column(deg_39_up)
names(deg_39_up)[1] <- "gene_id"
deg_39_up <- left_join(deg_39_up, conversion, by = "gene_id")
deg_39_up <- deg_39_up[,!(colnames(deg_39_up) %in% c("gene_id", "baseMean", "lfcSE", "stat"))]
deg_39_up <- dplyr::distinct(deg_39_up, gene_name, .keep_all = TRUE)
deg_39_up$gene_name <- toupper(deg_39_up$gene_name)
Limma_target_gene_overlap_DESEQ2_39 <- merge(deg_39_up, Limma_Drug_Targets_unique, by = "gene_name")
View(Limma_target_gene_overlap_DESEQ2_39)
write.csv(Limma_target_gene_overlap_DESEQ2_39, file = "C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\Limma_target_gene_overlap_DESEQ2_39.csv")
```


```{r}
rhub_launched_Drug_Targets <- read.csv("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\rhub_launcheddrugs_39.csv")
#separate gene targets into individual rows
rhub_launched_Drug_Targets <- tidyr::separate_rows(rhub_launched_Drug_Targets, Target, sep = ", ")
#remove drugs with empty 'target' field
rhub_launched_Drug_Targets <- dplyr::filter(rhub_launched_Drug_Targets, Target != "")
#Make new dataframe of just drug and gene name
Rhub_Launched_Drug_Targets_Unique <- data.frame(rhub_launched_Drug_Targets$pert, rhub_launched_Drug_Targets$Target)
#rename for merging function
Rhub_Launched_Drug_Targets_Unique <- setNames(Rhub_Launched_Drug_Targets_Unique, c("Drug", "gene_name"))
conversion <- as.data.frame(cts_39@elementMetadata)
conversion$gene_id <- gsub("\\..*", "", conversion$gene_id)
deg_39_up <- dsq2_39_up
deg_39_up <- tibble:::rownames_to_column(deg_39_up)
names(deg_39_up)[1] <- "gene_id"
deg_39_up <- left_join(deg_39_up, conversion, by = "gene_id")
deg_39_up <- deg_39_up[,!(colnames(deg_39_up) %in% c("gene_id", "baseMean", "lfcSE", "stat"))]
deg_39_up <- dplyr::distinct(deg_39_up, gene_name, .keep_all = TRUE)
deg_39_up$gene_name <- toupper(deg_39_up$gene_name)
rhub_target_gene_overlap_DESEQ2_39 <- merge(deg_39_up, Rhub_Launched_Drug_Targets_Unique, by = "gene_name")
View(rhub_target_gene_overlap_DESEQ2_39)
write.csv(rhub_target_gene_overlap_DESEQ2_39, file = "C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\rhub_target_gene_overlap_DESEQ2_39.csv")
```




#### Versions  
```{r}
R.Version()
```


```{r}
installed.packages()[names(sessionInfo()$otherPkgs), "Version"]
```
