---
title: "DESeq2_GSE69556"
author: "Ryan Brownlee"
date: '2022-07-13'
output: html_document
---
```{r setup, include=FALSE}
#filepath for code and data
knitr::opts_knit$set(root.dir = "/Users/eramsey/Downloads/PKD/kidney_drugrepurposing")
```
### Purpose  
This markdown uses nf-core for RNAseq alignment and DESeq2 DDS output  
Dataset: [GSE69556](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE69556)  
Processed data filepath: /data/project/lasseigne_lab/Lizzy/PKD_drugrepurposing_220324/  

#### Load in libraries
```{r message=FALSE, warning = FALSE}
library(DESeq2)
library(stringr)
library(apeglm)
library(pheatmap)
library(dplyr)
library(ggrepel)
library(signatureSearch)
library(ExperimentHub); library(rhdf5)
library(biomaRt)
library(viridis)
```

#### Custom functions  

__salmon2dds__: Salmon merged gene counts from nf-core RNAseq pipeline (results/star_salmon/salmon.merged.gene_counts.rds) and SRA metadata as inputs, function data wrangles and runs DESeq2, resulting in a dds object with design by genotype    

```{r}
#Salmon merged gene counts to dds results
salmon2dds_56 <- function(counts, metadata) {
  #count matrix in the sumarized experiment object
  cts <- counts@assays@data@listData[["counts"]]
  #round, salmon outputs have decimals
  cts <- round(cts)
  colnames(cts) <- str_sub(colnames(cts), 1, -2) ##ALTERATION just for GSE69556
  #remove version numbers from transcript id's
  rownames(cts) <- gsub("\\..*", "", rownames(cts))
  
  #remove all non-alphanumeric characters from Genotype
  metadata$Genotype <- str_replace_all(metadata$Genotype, "[^[:alnum:]]", "")
  #contrasts need to be factor levels
  metadata$Genotype <- as.factor(metadata$Genotype)
#print(rownames(metadata))
#print(colnames(cts))
  dds <- DESeqDataSetFromMatrix(cts, metadata, design = ~ Genotype)
return(dds)
}
```

__uplfc__ & __downlfc__: Formats deseq2 res() output to select for upregulated and downregulated genes by log2 fold change threshold > 2 and < -2.  

```{r}
uplfc <- function(res){
  #NAs removed first
  res <- res[!is.na(res$log2FoldChange),]
  #p-adjusted value cutoff 0.05 and log2fc greater than 2.0 and less than -2.0
  upfc <- res[res$log2FoldChange > 2.0,]
  #pull just logfold change values from DESeqResults object and name the values by associated gene
  up <- upfc$log2FoldChange
  #names(up) <- rownames(upfc)
  up <- cbind(Mouse.gene.stable.ID = rownames(upfc), UpLFC = up)
  
return(as.data.frame(up))
}
downlfc <- function(res){
  #NAs removed first
  res <- res[!is.na(res$log2FoldChange),]
  #p-adjusted value cutoff 0.05 and log2fc greater than 2.0 and less than -2.0
  downfc <- res[res$log2FoldChange < -2.0,]
  #pull just logfold change values from DESeqResults object and name the values by associated gene
  down <- downfc$log2FoldChange
  #names(down) <- rownames(downfc)
  down <- cbind(Mouse.gene.stable.ID = rownames(downfc), DownLFC = down)
  
return(as.data.frame(down))
}
```

__convertMouseGeneList__: Uses Ensembl human GRCh38.p13 and Ensembl mouse C57BL_6NJ_v1 genes and to find human orthologous genes (more details [here](https://docs.google.com/document/d/1jU3EOVaZXJMzxwH9SsssHoArz8pMdcfW8MSr8hbeE4Q/edit) ), and biomart to convert to human entrez  

```{r}
convertMouseGeneList <- function(lfc){
  #read in mose to human ens annotation
  ens_annot <- read.csv("/Users/eramsey/Downloads/PKD/kidney_drugrepurposing/data/annot_ens_humanmouse.csv")
  #conversion table
  ens_mouse2human <- merge(lfc, ens_annot, by = "Mouse.gene.stable.ID")
  #biomart to conver human ens to human entrez
  human <- useEnsembl("ensembl", dataset = "hsapiens_gene_ensembl", mirror = "useast")
  genes <- getBM(filters = "ensembl_gene_id", attributes = c("ensembl_gene_id","entrezgene_id"), values = ens_mouse2human$Gene.stable.ID, mart = human)
  
  names(ens_mouse2human) <- c("mouse_ensembl_gene_id", "LFC", "ensembl_gene_id")
  conv_tbl <- merge(genes, ens_mouse2human, by = "ensembl_gene_id")
return(conv_tbl)
}
```

__toplincs__: Uses uplfc and downlfc outputs to then find top 100 genes that are available in LINCS (best input for signature reversion)  
```{r message=FALSE}
eh <- ExperimentHub()
lincs <- eh[["EH3226"]]; lincs_expr <- eh[["EH3227"]]
lincs_genes <- rhdf5:::h5read(lincs, "rownames", drop=TRUE)
toplincs <- function(conv_tbl, direction){
  lincs_genes <- cbind(entrezgene_id = lincs_genes)
  conv_tbl <- merge(conv_tbl, lincs_genes, by = "entrezgene_id")
  conv_tbl$LFC <- as.numeric(conv_tbl$LFC)
  if(direction == "up"){
  top100 <- slice_max(conv_tbl, n = 100, order_by = LFC)
  }
  else{
    top100 <- slice_min(conv_tbl, n = 100, order_by = LFC)
  }
return(top100)
}
```

__lincsGenes__: Wraps convertMouseGeneList to map orthologs, and toplincs to filter for LINCS-available genes 
```{r}
lincsGenes <- function(lfc, direction){
  conv_tbl <- convertMouseGeneList(lfc)
  top100 <- toplincs(conv_tbl, direction = direction)
  return(top100)
}
```

### Read in data  
__Filepath to fastq's and metadata__: /data/project/lasseigne_lab/DATASET_dir/pkdDrugRepurposing/pkd2/GSE69556/SraRunTable.txt  
__Filepath to counts data__: /data/project/lasseigne_lab/Lizzy/PKD_drugrepurposing_220324/GSE69556_nfcore/results/star_salmon/salmon.merged.gene_counts.rds  
```{r}
cts_56 <- readRDS("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\GSE69556_salmonmerged_gene_counts.rds")
#cts_56 <- GSE69556_salmonmerged_gene_counts
#colnames(cts_56) <- str_sub(colnames(cts_56), 1, -2)
meta_56 <- read.csv("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\GSE69556_metadata.txt", row.names = 1)
```

#### Run DDS
```{r}
dds_56 <- salmon2dds_56(cts_56, meta_56)
str(dds_56)
```

#### Run DESeq2
```{r}
#relevel to make wildtype the reference instead of first alphabetical factor
dds_56$Genotype <- relevel(dds_56$Genotype, ref = "wildtype")
#run DESeq: estimates size factors, estimates dispersion, fits negative binomial GLM's and performs Wald stats
dds_56 <- DESeq(dds_56) #can also look at setting replaceOutliers lower for dealing with outliers 
```

#### Results  
Note: for LFC, if using apeglm for shrinkage estimate prior: If used in published research, please cite:  
Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for sequence count data: removing the noise and preserving large differences. Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895
```{r}
#extract results table with log2 fold changes, p values and adjusted p values for desired coefficients
res_56 <- results(dds_56, alpha = 0.05, contrast = c("Genotype", "Pkhd1CrePkd2FF", "wildtype"))
#Shrinkage of effect size (LFC estimates) is useful for visualization and ranking of genes. To shrink the LFC, we pass the ddds_nf_nf object to the function lfcShrink
resultsNames(dds_56) #prints coefficients
# "type" refers to different types of priors for shrinkage estimator that can be used: apeglm, ashr, or normal
## apeglm and ashr have been shown to have less bias than normal
resLFC_56 <- lfcShrink(dds_56, coef = c("Genotype_Pkhd1CrePkd2FF_vs_wildtype"), type="apeglm")
resLFC_56
```
DESeq2 results
```{r}
dsq2_56 <- as.data.frame(res_56)
dsq2_56_up <- dplyr::filter(dsq2_56, log2FoldChange > 2.0)
dsq2_56_down <- dplyr::filter(dsq2_56, log2FoldChange < -2.0)
```


```{r eval=FALSE, include=FALSE}
write.csv(res_56, file = "/Users/eramsey/Downloads/PKD/kidney_drugrepurposing/res/deseq2_outputs/deseq2_fullres_56.csv")
```


LFC cutoffs
```{r}
#NAs removed  p-adjusted value cutoff 0.05 and log2fc greater than 2.0 and less than -2.0
musdegs_56_UP <- uplfc(res_56)
musdegs_56_DOWN <- downlfc(res_56)
```

Map to human entrez and filter by LINCS genes
```{r}
hom_56_UP <- lincsGenes(musdegs_56_UP, "up")
hom_56_DOWN <-  lincsGenes(musdegs_56_DOWN, "down")
```


```{r eval=FALSE, include=FALSE}
write.csv(hom_56_UP, file = "/Users/eramsey/Downloads/PKD/kidney_drugrepurposing/res/deseq2_outputs/GSE69556_degs_UP_220421.csv")
write.csv(hom_56_DOWN, file = "/Users/eramsey/Downloads/PKD/kidney_drugrepurposing/res/deseq2_outputs/GSE69556_degs_DOWN_220421.csv")
```


Variance stabilized transformation  

rlog() may be better to use for this dataset -- better for high count variance by sample (in this case, one sample has over 2x the reads compared to the others) and for smaller datasets "if you have many samples (e.g. 100s), the rlog function might take too long, and so the vst function will be a faster choice. The rlog and VST have similar properties, but the rlog requires fitting a shrinkage term for each sample and each gene which takes time. See the DESeq2 paper for more discussion on the differences (Love, Huber, and Anders 2014)."  
```{r}
rld_56 <- rlog(dds_56, blind = FALSE) 
```


#### Visualizations  

PCA
```{r}
#regularized log transformation
plotPCA(rld_56, intgroup=c("Genotype"))
#label samples
plotPCA(rld_56, intgroup=c("Genotype")) + geom_label_repel(aes(label = rld_56@colData@rownames), box.padding   = 0.05, label.padding = 0.05, point.padding = 0.8, segment.color = 'grey50') +scale_colour_viridis_d() + theme_minimal() + theme(axis.title.x = element_text(size = 20))
```
Heatmap plotting:
```{r}
library(pheatmap)
#formatting for pheatmap
conversion3 <- as.data.frame(cts_56@elementMetadata)
conversion3$gene_id <- gsub("\\..*", "", conversion3$gene_id)
cts56_df <- as.data.frame(assays((rld_56)))
cts56_df <- tibble:::rownames_to_column(cts56_df, var = "gene_id")

conv_cts56 <- merge(cts56_df, conversion3, by = "gene_id")
conv_cts56 <- conv_cts56[,!(colnames(conv_cts56) %in% c("gene_id", "group", "group_name"))]
conv_cts56 <- dplyr::distinct(conv_cts56, gene_name, .keep_all = TRUE)
conv_cts56 <- tibble:::column_to_rownames(conv_cts56, "gene_name")

#Find out WT vs PDK2KO
rld_56$Genotype
cts_56$names
colnames(conv_cts56) <- c("wildtype", "wildtype", "wildtype", "wildtype", "PKD2", "PKD2", "PKD2")
#Convert 0 values to NA and remove from dataset
conv_cts56[conv_cts56==0] <- NA
conv_cts_fixed56 <- conv_cts56[complete.cases(conv_cts56),]

#need to convert the gene IDs here also
#genes <- order(resLFC_nf$log2FoldChange, decreasing=TRUE)[1:20]

#top20 genes w/ conv_cts
fixedtop20_56 <- order(conv_cts_fixed56$PKD2, decreasing = TRUE)[1:20]
#fixedtop20test <- order(conv_cts_fixed, decreasing= TRUE)[1:20]
pheatmap(conv_cts_fixed56[fixedtop20_56, ], cluster_rows = TRUE, cluster_cols = TRUE, show_rownames = TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "Top 20 Positive Genes '56", angle_col = 45)

#bottom 20 genes w/ conv_cts
fixedbottom20_56 <- order(conv_cts_fixed56$PKD2, decreasing = FALSE)[1:20]
pheatmap(conv_cts_fixed56[fixedbottom20_56, ], cluster_rows = TRUE, cluster_cols = FALSE, show_rownames = TRUE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "Bottom 20 Positive Genes '56", angle_col = 45)

#sort by variance
test56 <- apply(conv_cts56, 1, var)
#top 20 most variable genes
top56 <- names(sort(test56, decreasing = TRUE))[1:20]

 #20 most varied genes
#estimate variance for each row 
pheatmap(conv_cts56[top56,], cluster_rows=TRUE, show_rownames=TRUE, cluster_cols=FALSE, color = viridis(n= 10000, alpha = 1, begin = 0, end = 1, option = "viridis"), border_color = "NA", main = "20 Most Varied Genes Cystic P21", angle_col = 45) #, annotation_col=meta_39$Genotype)
```
#volcano plots
```{r}
Vol56 <- as.data.frame(res_56)
TopT_test56 <- tibble:::rownames_to_column(Vol56)
names(TopT_test56)[1] <- "gene_id"
TopT_finish56 <- left_join(TopT_test56, conversion3, by = "gene_id")
TopT_finish56 <- na.omit(TopT_finish56)
TopT_finish56 <- TopT_finish56[,!(colnames(TopT_finish56) %in% c("gene_id", "baseMean", "lfcSE", "stat"))]
TopT_finish56 <- dplyr::distinct(TopT_finish56, gene_name, .keep_all = TRUE)
TopT_finish56 <- tibble:::column_to_rownames(TopT_finish56, "gene_name")
topt_finish_wack <- dplyr::distinct(TopT_finish, gene_name, .keep_all = TRUE)

View(TopT_finish56)
#volcano plot w/ strict pvalue cutoff
EnhancedVolcano(TopT_finish56, lab = rownames(TopT_finish56), x = 'log2FoldChange', y = 'pvalue', pCutoff = 10e-32, FCcutoff = 0.5, pointSize = 3.0, labSize = 6.0, col = c("#440154", "#21918c", "#5ec962", "#fde725"))
#basic volcano plot
EnhancedVolcano(TopT_finish56, lab = rownames(TopT_finish56), x = 'log2FoldChange', y = 'pvalue', pointSize = 3.0, labSize = 6.0, col = c("#440154", "#21918c", "#5ec962", "#fde725"))
```
```{r}
deg_56_up <- dsq2_56_up
deg_56_up <- tibble:::rownames_to_column(deg_56_up)
names(deg_56_up)[1] <- "gene_id"
deg_56_up <- left_join(deg_56_up, conversion, by = "gene_id")
deg_56_up <- deg_56_up[,!(colnames(deg_56_up) %in% c("gene_id", "baseMean", "lfcSE", "stat"))]
deg_56_up <- dplyr::distinct(deg_56_up, gene_name, .keep_all = TRUE)
#deg_56_up <- tibble:::column_to_rownames(deg_56_up, "gene_name")
# Read in Drug Targets
gene_targets <- read.csv("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\gene_drug_targets_39.csv")
gene_targets_unique <- data.frame(gene_targets$pert, gene_targets$Target)
gene_targets_unique <- setNames(gene_targets_unique, c("Drug", "gene_name"))
# deg_39_up is in lowercase and gene_targets_unique_df is uppercase. must fix!
deg_56_up$gene_name <- toupper(deg_56_up$gene_name)
#now fixed, merge for similarities in targets
target_gene_overlap_DESEQ2_56 <- merge(deg_56_up, gene_targets_unique, by = "gene_name")
View(target_gene_overlap_DESEQ2_56)
write.csv(target_gene_overlap_DESEQ2_56, file = "C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\target_gene_overlap_DESEQ2_56.csv")

```

```{r}
Limma_Drug_Targets <- read.csv("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\rhub_limma_launcheddrugs_39.csv")
#separate gene targets into individual rows
Limma_Drug_Targets <- tidyr::separate_rows(Limma_Drug_Targets, Target, sep = ", ")
#remove drugs with empty 'target' field
Limma_Drug_Targets <- dplyr::filter(Limma_Drug_Targets, Target != "")
#Make new dataframe of just drug and gene name
Limma_Drug_Targets_unique <- data.frame(Limma_Drug_Targets$pert, Limma_Drug_Targets$Target)
#rename for merging function
Limma_Drug_Targets_unique <- setNames(Limma_Drug_Targets_unique, c("Drug", "gene_name"))
conversion <- as.data.frame(cts_19@elementMetadata)
conversion$gene_id <- gsub("\\..*", "", conversion$gene_id)
deg_56_up <- dsq2_56_up
deg_56_up <- tibble:::rownames_to_column(deg_56_up)
names(deg_56_up)[1] <- "gene_id"
deg_56_up <- left_join(deg_56_up, conversion, by = "gene_id")
deg_56_up <- deg_56_up[,!(colnames(deg_56_up) %in% c("gene_id", "baseMean", "lfcSE", "stat"))]
deg_56_up <- dplyr::distinct(deg_56_up, gene_name, .keep_all = TRUE)
deg_56_up$gene_name <- toupper(deg_56_up$gene_name)
Limma_target_gene_overlap_DESEQ2_56 <- merge(deg_56_up, Limma_Drug_Targets_unique, by = "gene_name")
View(Limma_target_gene_overlap_DESEQ2_56)
write.csv(Limma_target_gene_overlap_DESEQ2_56, file = "C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\Limma_target_gene_overlap_DESEQ2_56.csv")
```


```{r}
rhub_launched_Drug_Targets <- read.csv("C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\rhub_launcheddrugs_39.csv")
#separate gene targets into individual rows
rhub_launched_Drug_Targets <- tidyr::separate_rows(rhub_launched_Drug_Targets, Target, sep = ", ")
#remove drugs with empty 'target' field
rhub_launched_Drug_Targets <- dplyr::filter(rhub_launched_Drug_Targets, Target != "")
#Make new dataframe of just drug and gene name
Rhub_Launched_Drug_Targets_Unique <- data.frame(rhub_launched_Drug_Targets$pert, rhub_launched_Drug_Targets$Target)
#rename for merging function
Rhub_Launched_Drug_Targets_Unique <- setNames(Rhub_Launched_Drug_Targets_Unique, c("Drug", "gene_name"))
conversion <- as.data.frame(cts_56@elementMetadata)
conversion$gene_id <- gsub("\\..*", "", conversion$gene_id)
deg_56_up <- dsq2_56_up
deg_56_up <- tibble:::rownames_to_column(deg_56_up)
names(deg_56_up)[1] <- "gene_id"
deg_56_up <- left_join(deg_56_up, conversion, by = "gene_id")
deg_56_up <- deg_56_up[,!(colnames(deg_56_up) %in% c("gene_id", "baseMean", "lfcSE", "stat"))]
deg_56_up <- dplyr::distinct(deg_56_up, gene_name, .keep_all = TRUE)
deg_56_up$gene_name <- toupper(deg_56_up$gene_name)
rhub_target_gene_overlap_DESEQ2_56 <- merge(deg_56_up, Rhub_Launched_Drug_Targets_Unique, by = "gene_name")
View(rhub_target_gene_overlap_DESEQ2_56)
write.csv(rhub_target_gene_overlap_DESEQ2_56, file = "C:\\Users\\mattr\\Documents\\Lasseigne_Lab_KURE\\TFEA_Lasseigne_Lab_Brownlee\\rhub_target_gene_overlap_DESEQ2_56.csv")
```








#### Versions  
```{r}
R.Version()
```


```{r}
installed.packages()[names(sessionInfo()$otherPkgs), "Version"]
```
